org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -7,6 +7,16 @@
 
 namespace MixedReality.Toolkit.Input
 {
+    /// <summary>
+    /// This interface is used to update the front plate and rounded rect of a pressable
+    /// button if they are flagged as dynamic (based on proximity), it is needed to prevent
+    /// a circular reference between MRTK Input and MRTK UX Core Scripts packages.
+    /// </summary>
+    public interface IPressableButtonForNearInteractionModeDetector
+    {
+        public void UpdateFrontPlateAndRoundedRectIfDynamic(bool enable);
```

Comment:
Thank you for the suggestion but it won’t work as how NearInteractionModeDetector works right now because it only keeps track of the last interactor triggering the near interaction.  For example, I created this test:
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/37ef8075-6b6d-4e0e-85ff-dc0c9c4a34f7)
Which does this:
1. Creates a dialog with a button
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/19e978a4-17e8-4937-8893-b12b62534493)
2. Creates a right hand and touches the button
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/4a7a8e77-4090-45c6-8f9f-d73f1baf66db)
3. Creates a left hand and touches the button
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/dba709af-3a14-415b-a73d-3236959a84cf)
At all steps the NearInteractors count remain the same, the only difference is that the right hand interactor reference gets replaced with the left hand interactor reference.
I am currently investigating if there is already a way to query or if there is a data structure that holds all active interactors but I am new to this part of the code (if you know of any please let me know).  Otherwise, I will have to add some way to keep track of that but I was hoping to keep this PR simple.

Reply:
The last commit adds the functionality to support multiple interactors when dynamically enabling/disabling buttons components. I also added a runtime test that verifies the new behavior.
I believe the commit addresses this issue so resolving this conversation, feel free to re-open if needed.